x-pack/filebeat/input/awss3/input.go

```diff
@@ -46,17 +49,41 @@ func (im *s3InputManager) Create(cfg *conf.C) (v2.Input, error) {
 		return nil, fmt.Errorf("initializing AWS config: %w", err)
 	}
 
-	// The awsConfig now contains the region from the credential profile or default region
-	// if the region is explicitly set in the config, then it wins
-	if config.RegionName != "" {
+	if config.AccessPointARN != "" {
+		// When using the access point ARN, requests must be directed to the
+		// access point hostname. The access point hostname takes the form
+		// AccessPointName-AccountId.s3-accesspoint.Region.amazonaws.com
+		arnParts := strings.Split(config.AccessPointARN, ":")
+		region := arnParts[3]
+		accountID := arnParts[4]
+		accessPointName := strings.Split(arnParts[5], "/")[1]
+
+		// Construct the endpoint for the Access Point
+		endpoint := fmt.Sprintf("%s-%s.s3-accesspoint.%s.amazonaws.com", accessPointName, accountID, region)
```

Comment:
Seems like this is important for this feature. Could we move the endpoint calculation into a separate function and add some unit tests too?

Reply:
Sure, added at 57bcba71e0d086623c9cac07918d0f8435b90d02

Reply:
Thank you!